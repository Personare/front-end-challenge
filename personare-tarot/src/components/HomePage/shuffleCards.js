// export const keyGen = ()=> { return( Math.random().toString(25).substring(2,15)+
//     Math.random().toString(25).substring(2,15) )};



// export const handleClickFn = (id) => {
//     if (handleClick) {
//       handleClick(id);
//     }
//   };

// export const sortArray = (cartas) => {
//     return cartas.sort(() => Math.random() - 0,5)
// };

// export const regenerateSortArray = (cards) => {
//     return sortArray(regenerateCard(cards))
// }

// console.log(regenerateSortArray)