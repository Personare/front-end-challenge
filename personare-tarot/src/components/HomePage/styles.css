
.card_content{
  margin: 5px;
    display: flex;
    flex-direction: row;
    position:relative;
    width: 100px;
    height: 100px;
    transform-style: preserve-3d;
  transition: transform 600ms ease-in-out;
}


.card_image {
    display: grid;
    grid-template-columns: repeat(5, 1fr);
    grid-template-rows: repeat(5,1fr);
    text-align: center;
    padding-top: 0px;
    min-width: 100%;
    border-radius: 50px;
}

.button-start{
  margin: 10px;
  display: inline-block;
  padding: 15px 25px;
  font-size: 24px;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  outline: none;
  color: #fff;
  background-color:	#FFB6C1;
  border: none;
  border-radius: 15px;
  box-shadow: 0 9px #999;
}



