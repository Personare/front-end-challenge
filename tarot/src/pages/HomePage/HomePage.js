import React from 'react'


const HomePage =()=>{
    return(
        <div>
            <h1>Tela inicial</h1>
        </div>
    )
}

export default HomePage