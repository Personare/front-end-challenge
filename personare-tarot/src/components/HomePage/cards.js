import React, { useState } from "react";
import axios from "axios";
import { useEffect } from "react";
import './styles.css';


export const Cards = () => {

    const [cartas, setCartas] = useState([])
    const [randomCard, setRandomCard] = useState(false)
    const [imageCard, setImageCard] = useState("")

    useEffect(() => {
        axios
            .get("https://raw.githubusercontent.com/Personare/front-end-challenge/master/tarot.json")
            .then((response) => {
                setCartas(response.data)
            }).catch((error) => { console.log(error) })
    }, [])

    const shuffle = () => {
        const shuffleCards = Math.floor(Math.random() * 78)
        setImageCard(cartas.cards[shuffleCards].image)
        setRandomCard(!false)

    }
    const mapCartas = cartas.cards && cartas.cards.map((card) => {
        return (

            < div onClick={shuffle} key={card.name} className="card">
                <img src={cartas.imageBackCard} alt="errorImage" />
            </div>
        )
    }) 
   
    return (
        <>
            <button className="button-start" onClick={()=>window.location.reload(false)}>Iniciar Jogo!</button>
            <div className="card_content">
            <div className="card_image">
                {randomCard ? (
                    <img src={cartas.imagesUrl + imageCard} alt="errorImage" />
                ) : (<div>
                    {mapCartas}
                </div>)}
            </div>
            </div>
        </>

    )
}
