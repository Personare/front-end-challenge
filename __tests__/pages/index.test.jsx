import React from 'react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import {
    render,
    screen,
    fireEvent,
    waitForElementToBeRemoved,
    waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import 'whatwg-fetch';

import HomePage from '../../pages/index';
import tarotData from '../tarotMock.json';

// Mock do shuffle aleatório do Card Manager para que a ordem seja sempre o inverso do original.
jest.mock('../../components/utils', () => {
    return {
        __esModule: true,
        shuffleArrayElements: (array) => {
            const cards = [...array];
            return cards.reverse();
        },
    };
});

const server = setupServer();

beforeAll(() => {
    server.listen();
});

afterEach(() => {
    server.resetHandlers();
});

afterAll(() => {
    server.close();
    jest.clearAllMocks();
});

describe('HomePage', () => {
    it('Snapshot do componente enquanto aguarda a resposta da API.', async () => {
        server.use(
            rest.get(process.env.NEXT_PUBLIC_API, (req, res, ctx) => {
                return res(ctx.status(200), ctx.json(tarotData));
            })
        );

        const { asFragment } = render(<HomePage />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Snapshot do componente caso a API retorne algum erro', async () => {
        server.use(
            rest.get(process.env.NEXT_PUBLIC_API, (req, res, ctx) => {
                return res(ctx.status(500), ctx.json({ msg: 'Mock' }));
            })
        );

        const { asFragment } = render(<HomePage />);

        // Aguarda a animação do spinner acabar.
        await waitFor(
            () => {
                const spinnerPhase2 = screen.getByTestId('spinner-2');
                fireEvent.animationEnd(spinnerPhase2);
            },
            { timeout: 3000 }
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Snapshot do componente após o carregamento da API e antes de iniciar o jogo.', async () => {
        server.use(
            rest.get(process.env.NEXT_PUBLIC_API, (req, res, ctx) => {
                return res(ctx.status(200), ctx.json(tarotData));
            })
        );

        const { asFragment } = render(
            <main>
                <HomePage />
            </main>
        );

        // Aguarda o spinner sair da tela após receber os dados da API.
        await waitForElementToBeRemoved(() => screen.getByTestId('spinner-1'), {
            timeout: 3000,
        });

        // Esperar as cartas terminarem sua animação.
        await waitFor(() => {
            expect(screen.getByTestId('1-0')).toBeInTheDocument();
        });

        expect(asFragment()).toMatchSnapshot();
    });

    it('Snapshot do componente depois de iniciar o jogo.', async () => {
        server.use(
            rest.get(process.env.NEXT_PUBLIC_API, (req, res, ctx) => {
                return res(ctx.status(200), ctx.json(tarotData));
            })
        );

        const { asFragment } = render(
            <main>
                <HomePage />
            </main>
        );

        await waitForElementToBeRemoved(() => screen.getByTestId('spinner-1'), {
            timeout: 3000,
        });

        const btnStartGame = screen.getByRole('button', {
            name: 'Iniciar o Jogo',
        });
        fireEvent.click(btnStartGame);

        // Ativando os eventos
        const card1Phase2 = screen.getByTestId('2-0');
        const card2Phase2 = screen.getByTestId('2-1');
        const card3Phase2 = screen.getByTestId('2-2');
        fireEvent.animationEnd(card1Phase2);
        fireEvent.animationEnd(card2Phase2);
        fireEvent.animationEnd(card3Phase2);

        const card1Phase3 = screen.getByTestId('3-0');
        const card2Phase3 = screen.getByTestId('3-1');
        const card3Phase3 = screen.getByTestId('3-2');
        fireEvent.animationEnd(card1Phase3);
        fireEvent.animationEnd(card2Phase3);
        fireEvent.animationEnd(card3Phase3);

        expect(asFragment()).toMatchSnapshot();
    });

    it('Snapshot do componente depois de iniciar o jogo e selecionar uma carta.', async () => {
        server.use(
            rest.get(process.env.NEXT_PUBLIC_API, (req, res, ctx) => {
                return res(ctx.status(200), ctx.json(tarotData));
            })
        );

        const { asFragment } = render(
            <main>
                <HomePage />
            </main>
        );
        await waitForElementToBeRemoved(() => screen.getByTestId('spinner-1'), {
            timeout: 3000,
        });

        const btnStartGame = screen.getByRole('button', {
            name: 'Iniciar o Jogo',
        });
        fireEvent.click(btnStartGame);

        // Ativando os eventos
        const card1Phase2 = screen.getByTestId('2-0');
        const card2Phase2 = screen.getByTestId('2-1');
        const card3Phase2 = screen.getByTestId('2-2');
        fireEvent.animationEnd(card1Phase2);
        fireEvent.animationEnd(card2Phase2);
        fireEvent.animationEnd(card3Phase2);

        const card1Phase3 = screen.getByTestId('3-0');
        const card2Phase3 = screen.getByTestId('3-1');
        const card3Phase3 = screen.getByTestId('3-2');
        fireEvent.animationEnd(card1Phase3);
        fireEvent.animationEnd(card2Phase3);
        fireEvent.animationEnd(card3Phase3);

        const card1Phase4 = screen.getByTestId('4-0');
        fireEvent.click(card1Phase4);

        expect(asFragment()).toMatchSnapshot();
    });
});
