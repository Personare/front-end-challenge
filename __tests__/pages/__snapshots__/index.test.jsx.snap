// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Snapshot do componente após o carregamento da API e antes de iniciar o jogo. 1`] = `
<DocumentFragment>
  .c1 {
  font-family: inherit;
  font-size: 1em;
  line-height: 1em;
  font-weight: bold;
  padding: 1em 2em;
  cursor: pointer;
  border-radius: 50px;
  color: #0d0d0d;
  background-color: #f2ea7e;
  border: 2px solid #f2ea7e;
}

.c1:hover {
  border: 2px solid #d1c848 !important;
}

.c3 {
  width: 150px;
  height: 300px;
}

.c3:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: scale 200ms;
  transition: scale 200ms;
}

.c4 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/162x341/arcano1.jpg);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: fCDSQb;
  animation: fCDSQb;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c5 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/162x341/arcano2.jpg);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: fCDSQb;
  animation: fCDSQb;
  -webkit-animation-delay: 20ms;
  animation-delay: 20ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c6 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/162x341/arcano3.jpg);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: fCDSQb;
  animation: fCDSQb;
  -webkit-animation-delay: 40ms;
  animation-delay: 40ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c0 {
  margin: 5em 0.5em;
  color: #0d0d0d;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  row-gap: 20px;
  -webkit-column-gap: 5px;
  column-gap: 5px;
  margin-bottom: 20px;
}

<main>
    <div
      class="c0"
    >
      <h1>
        Katreque's Tarot Game
      </h1>
      <button
        class="c1"
      >
        Iniciar o Jogo
      </button>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3 c4"
        data-testid="1-0"
        mode="both"
      />
      <div
        class="c3 c5"
        data-testid="1-1"
        mode="both"
      />
      <div
        class="c3 c6"
        data-testid="1-2"
        mode="both"
      />
    </div>
  </main>
</DocumentFragment>
`;

exports[`HomePage Snapshot do componente caso a API retorne algum erro 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-animation-name: fCDSQb;
  animation-name: fCDSQb;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  margin-top: 50px;
  font-size: 1.25em;
  -webkit-animation-name: fCDSQb;
  animation-name: fCDSQb;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 500ms;
  animation-delay: 500ms;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c3 {
  margin-bottom: 10px;
  font-weight: bold;
}

.c0 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="c0"
  >
    <div
      class="c1"
      data-testid="spinner-3"
    >
      <span
        style="background-color: rgb(245, 33, 50); width: 20px; height: 20px; border-radius: 100%; display: inline-block; position: relative; animation-fill-mode: both; animation: react-spinners-RotateLoader-rotate 5s 0s infinite cubic-bezier(0.7, -0.13, 0.22, 0.86);"
      >
        <span
          style="background-color: rgb(245, 33, 50); width: 20px; height: 20px; border-radius: 100%; opacity: 0.8; position: absolute; top: 0px; left: -28px;"
        />
        <span
          style="background-color: rgb(245, 33, 50); width: 20px; height: 20px; border-radius: 100%; opacity: 0.8; position: absolute; top: 0px; left: 28px;"
        />
      </span>
    </div>
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        Houston, we have a problem...
      </span>
      <small>
        Recarregue a página e tente novamente.
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HomePage Snapshot do componente depois de iniciar o jogo e selecionar uma carta. 1`] = `
<DocumentFragment>
  .c18 {
  font-family: inherit;
  font-size: 1em;
  line-height: 1em;
  font-weight: bold;
  padding: 1em 2em;
  cursor: pointer;
  border-radius: 50px;
  color: white;
  background-color: #0d0d0d;
  border: 2px solid #0d0d0d;
}

.c18:hover {
  border: 2px solid #6e6e6e !important;
}

.c12 {
  -webkit-animation: kcCFSJ;
  animation: kcCFSJ;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c12.close-modal {
  opacity: 0;
  visibility: collapse;
}

.c19 {
  z-index: 3;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(33,58,74,0.75);
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

.c13 {
  z-index: 4;
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 600px;
  max-width: 100%;
  max-height: 100%;
  background-color: white;
  border-radius: 15px;
  box-shadow: 0px 0px 0px 2px rgba(0,0,0,0.1);
  -webkit-animation-delay: 0.75s;
  animation-delay: 0.75s;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 10px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 10px;
}

.c16 {
  color: #0d0d0d;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  margin: 10px;
  padding: 5px;
}

.c17 {
  margin: 0.25em 0em;
}

.c3 {
  width: 150px;
  height: 300px;
}

.c3:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: scale 200ms;
  transition: scale 200ms;
}

.c4 {
  position: relative;
}

.c5 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/back-blue-card.png);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: ddxGlr;
  animation: ddxGlr;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c9 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/back-blue-card.png);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: fCDSQb;
  animation: fCDSQb;
  -webkit-animation-delay: 20ms;
  animation-delay: 20ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c11 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/back-blue-card.png);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: fCDSQb;
  animation: fCDSQb;
  -webkit-animation-delay: 40ms;
  animation-delay: 40ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c7 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/162x341/arcano3.jpg);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: ddxGlr;
  animation: ddxGlr;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c6 {
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-animation-duration: 0.75s;
  animation-duration: 0.75s;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.c8 {
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-animation-duration: 0.75s;
  animation-duration: 0.75s;
  -webkit-animation-direction: reverse;
  animation-direction: reverse;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.c10 {
  cursor: pointer;
  z-index: 2;
}

.c0 {
  margin: 5em 0.5em;
  color: #0d0d0d;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 1.5em;
  font-weight: bold;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  row-gap: 20px;
  -webkit-column-gap: 5px;
  column-gap: 5px;
  margin-bottom: 20px;
}

<main>
    <div
      class="c0"
    >
      <p
        class="c1"
      >
        Selecione uma carta
      </p>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3 c4"
        data-testid="5-0"
      >
        <div
          class="c3 c5 c6"
          mode="both"
        />
        <div
          class="c3 c7 c8"
          direction="reverse"
          mode="both"
        />
      </div>
      <div
        class="c3 c9 c10"
        data-testid="4-1"
        mode="both"
      />
      <div
        class="c3 c11 c10"
        data-testid="4-2"
        mode="both"
      />
    </div>
    <div
      data-testid="card-expanded"
    >
      <div
        class="c12 c13"
      >
        <div
          class="c14"
        >
          <div
            class="c15"
          >
            <img
              alt="A IMPERATRIZ"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fdkw5ssdvaqf8l.cloudfront.net%2Fstatic%2Fpsr%2Fbr%2Fframework%2Fyii%2Fimages%2Fcontent%2Fpt-br%2Fproduct%2Ftarot%2Fmarselha%2F162x341%2Farcano3.jpg&w=384&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fdkw5ssdvaqf8l.cloudfront.net%2Fstatic%2Fpsr%2Fbr%2Fframework%2Fyii%2Fimages%2Fcontent%2Fpt-br%2Fproduct%2Ftarot%2Fmarselha%2F162x341%2Farcano3.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fdkw5ssdvaqf8l.cloudfront.net%2Fstatic%2Fpsr%2Fbr%2Fframework%2Fyii%2Fimages%2Fcontent%2Fpt-br%2Fproduct%2Ftarot%2Fmarselha%2F162x341%2Farcano3.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </div>
          <div
            class="c16"
          >
            <h2
              class="c17"
            >
              A IMPERATRIZ
            </h2>
            <p>
              Mussum Ipsum, cacilds vidis litro abertis. Viva Forevis aptent taciti sociosqu ad litora torquent. Admodum accumsan disputationi eu sit. Vide electram sadipscing et per.
            </p>
            <button
              class="c18"
            >
              Mais uma?
            </button>
          </div>
        </div>
      </div>
      <div
        class="c12 c19"
      />
    </div>
  </main>
</DocumentFragment>
`;

exports[`HomePage Snapshot do componente depois de iniciar o jogo. 1`] = `
<DocumentFragment>
  .c3 {
  width: 150px;
  height: 300px;
}

.c3:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: scale 200ms;
  transition: scale 200ms;
}

.c4 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/back-blue-card.png);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: fCDSQb;
  animation: fCDSQb;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c6 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/back-blue-card.png);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: fCDSQb;
  animation: fCDSQb;
  -webkit-animation-delay: 20ms;
  animation-delay: 20ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c7 {
  background-image: url(https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/back-blue-card.png);
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  -webkit-animation: fCDSQb;
  animation: fCDSQb;
  -webkit-animation-delay: 40ms;
  animation-delay: 40ms;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c5 {
  cursor: pointer;
  z-index: 2;
}

.c0 {
  margin: 5em 0.5em;
  color: #0d0d0d;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 1.5em;
  font-weight: bold;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  row-gap: 20px;
  -webkit-column-gap: 5px;
  column-gap: 5px;
  margin-bottom: 20px;
}

<main>
    <div
      class="c0"
    >
      <p
        class="c1"
      >
        Selecione uma carta
      </p>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3 c4 c5"
        data-testid="4-0"
        mode="both"
      />
      <div
        class="c3 c6 c5"
        data-testid="4-1"
        mode="both"
      />
      <div
        class="c3 c7 c5"
        data-testid="4-2"
        mode="both"
      />
    </div>
  </main>
</DocumentFragment>
`;

exports[`HomePage Snapshot do componente enquanto aguarda a resposta da API. 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-animation-name: fCDSQb;
  animation-name: fCDSQb;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c0 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="c0"
  >
    <div
      class="c1"
      data-testid="spinner-1"
    >
      <span
        style="background-color: rgb(75, 131, 166); width: 20px; height: 20px; border-radius: 100%; display: inline-block; position: relative; animation-fill-mode: both; animation: react-spinners-RotateLoader-rotate 2s 0s infinite cubic-bezier(0.7, -0.13, 0.22, 0.86);"
      >
        <span
          style="background-color: rgb(75, 131, 166); width: 20px; height: 20px; border-radius: 100%; opacity: 0.8; position: absolute; top: 0px; left: -28px;"
        />
        <span
          style="background-color: rgb(75, 131, 166); width: 20px; height: 20px; border-radius: 100%; opacity: 0.8; position: absolute; top: 0px; left: 28px;"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
